### PR TITLE
Convert from groovy to junit

### PR DESCRIPTION
**Pull request by cssjessica**
## Description of Changes

The following spock groovy test were converted to Junit: PointStreamSpec and Nc4IospSpec from Issue https://github.com/Unidata/netcdf-java/issues/1188.

## PR Checklist

- [x] Link to any issues that the PR addresses
- [ ] Add labels
- [x] Open as a [draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/)
       until ready for review
- [x] Make sure GitHub tests pass
- [x] Mark PR as "Ready for Review"

### CODE REVIEW

File: cdm/core/src/test/java/ucar/nc2/ft/point/remote/TestPointStream.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import ucar.nc2.constants.FeatureType;
+import ucar.nc2.ft.FeatureDatasetFactoryManager;
+import ucar.nc2.ft.FeatureDatasetPoint;
+import ucar.nc2.ft.NoFactoryFoundException;
+import ucar.nc2.ft.PointFeatureCollection;
+import ucar.nc2.ft.point.FlattenedDatasetPointCollection;
+import ucar.nc2.ft.point.PointTestUtil;
+import ucar.nc2.ft.point.remote.PointCollectionStreamLocal;
```
Comment (haileyajohnson) [inline cdm/core/src/test/java/ucar/nc2/ft/point/remote/TestPointStream.java]: These two imports are unused (PointCollectionStreamLocal and PointStream)

File: cdm/core/src/test/java/ucar/nc2/ft/point/remote/TestPointStream.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    this.location = location;
+  }
+
+  @Test
+  public void roundTrip() throws IOException, NoFactoryFoundException {
+
+
+    File outFile = temporaryFolder.newFile();
+    try (FeatureDatasetPoint fdPoint =
+        (FeatureDatasetPoint) FeatureDatasetFactoryManager.open(FeatureType.ANY_POINT, location, null)) {
```
Comment (haileyajohnson) [inline cdm/core/src/test/java/ucar/nc2/ft/point/remote/TestPointStream.java]: This method is deprecated, the new method signature takes and error log formatter (which can just be null for the sake of using the current API). This test also then won't throw the `NoFactoryFoundException`

File: cdm/core/src/test/java/ucar/nc2/ft/point/remote/TestPointStream.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+  String location;
+
+  public TestPointStream(String location) {
+    this.location = location;
+  }
+
+  @Test
+  public void roundTrip() throws IOException, NoFactoryFoundException {
+
```
Comment (haileyajohnson) [inline cdm/core/src/test/java/ucar/nc2/ft/point/remote/TestPointStream.java]: Delete extra whitespace :)

File: cdm/core/src/test/java/ucar/nc2/ft/point/remote/TestPointStream.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      PointFeatureCollection origPointCol = new FlattenedDatasetPointCollection(fdPoint);
+      PointStream.write(origPointCol, outFile);
+      PointFeatureCollection roundTrippedPointCol = new PointCollectionStreamLocal(outFile);
+
+      assertThat(PointTestUtil.equals(origPointCol, roundTrippedPointCol)).isTrue();
+    }
+
+  }
+
```
Comment (haileyajohnson) [inline cdm/core/src/test/java/ucar/nc2/ft/point/remote/TestPointStream.java]: Delete extra whitespace :)

File: netcdf4/src/test/groovy/ucar/nc2/jni/netcdf/Nc4IospSpec.groovy [hunk 5]
```diff
@@ -1,46 +0,0 @@
...
- * @since 2016-12-27
- */
-class Nc4IospSpec extends Specification {
-    private static final Logger logger = LoggerFactory.getLogger(Nc4IospSpec)
-    
-    def setup() {
-        // Ignore this class's tests if NetCDF-4 isn't present.
-        // We're using setup() because it shows these tests as being ignored.
-        // setupSpec() shows them as *non-existent*, which is not what we want.
-        Assume.assumeTrue("NetCDF-4 C library not present.", Nc4Iosp.isClibraryPresent());
```
Comment (haileyajohnson) [inline netcdf4/src/test/groovy/ucar/nc2/jni/netcdf/Nc4IospSpec.groovy:22]: I think we still need to check that the C library is present for these tests. You can follow what this test does: https://github.com/Unidata/tds/blob/13017bd6dfd9f5ae270bf2e524d8d7340592742b/tds/src/test/java/thredds/server/ncss/view/dsg/DsgSubsetWriterTest.java#L47

**Review by haileyajohnson — CHANGES_REQUESTED** [review]

File: netcdf4/src/test/groovy/ucar/nc2/jni/netcdf/Nc4IospSpec.groovy (hunk 5 above)
Comment (tdrwenski) [reply netcdf4/src/test/groovy/ucar/nc2/jni/netcdf/Nc4IospSpec.groovy:22]: If I understand the comment correctly they wanted the tests to be ignored if the C Library isn't present. But these tests just test that no exception is thrown when a method is called, which should be the case whether or not the library exists, so I don't think that assumption is really needed

Comment (haileyajohnson) [reply netcdf4/src/test/groovy/ucar/nc2/jni/netcdf/Nc4IospSpec.groovy:22]: wait, what are these even meant to be testing?

Comment (tdrwenski) [reply netcdf4/src/test/groovy/ucar/nc2/jni/netcdf/Nc4IospSpec.groovy:22]: My understanding was that at some point if you didn't have the netcdf-c library present then calling these functions caused a null pointer exception. Then they made these tests and added a null check to fix that. So they only check that there is no null pointer exception thrown, but yeah not the most useful tests of all time 😆

Comment (haileyajohnson) [reply netcdf4/src/test/groovy/ucar/nc2/jni/netcdf/Nc4IospSpec.groovy:22]: lol well now they don't do anything cuz the functions return right away 🤷

Comment (tdrwenski) [reply netcdf4/src/test/groovy/ucar/nc2/jni/netcdf/Nc4IospSpec.groovy:22]: yeah it's not a super useful test... do you still want to check for the c library or leave as is?

File: cdm/core/src/test/java/ucar/nc2/ft/point/remote/TestPointStream.java (hunk 2 above)
Comment (cssjessica) [reply cdm/core/src/test/java/ucar/nc2/ft/point/remote/TestPointStream.java]: The method was updated.

File: netcdf4/src/test/groovy/ucar/nc2/jni/netcdf/Nc4IospSpec.groovy (hunk 5 above)
Comment (cssjessica) [reply netcdf4/src/test/groovy/ucar/nc2/jni/netcdf/Nc4IospSpec.groovy:22]: We ended up adding the check for the C library

**Review by haileyajohnson — APPROVED** [review]

**Pull request merged**